plugins/for_testing_only/test_hooks/test_hooks.cc

```diff
@@ -0,0 +1,319 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+Regression test code for TS API HTTP hooks.  The code assumes there will only be one active transaction at a time.  It
+verifies the event data parameter to the continuations triggered by the hooks is correct.
+*/
+
+#include <fstream>
+#include <cstdlib>
+
+#include <ts/ts.h>
+
+// TSReleaseAssert() doesn't seem to produce any logging output for a debug build, so do both kinds of assert.
+//
+#define ALWAYS_ASSERT(EXPR) \
+  {                         \
+    TSAssert(EXPR);         \
+    TSReleaseAssert(EXPR);  \
+  }
+
+namespace
+{
+#define PINAME "test_hooks"
+char PIName[] = PINAME;
+
+// NOTE:  It's important to flush this after writing so that a gold test using this plugin can examine the log before TS
```

Comment:
Can't it be set to no-buffering mode?

Reply:
[Using setbuf/pubsetbuf](http://www.cplusplus.com/reference/fstream/filebuf/setbuf/).

Reply:
May take me a few days to figure this out.

Reply:
[This might help](https://stackoverflow.com/questions/16605233/how-to-disable-buffering-on-a-stream).